File: listaExercicios/Exceções/Excecao.java
class SaldoInsuficienteException extends Exception {
    public SaldoInsuficienteException(String message) {
        super(message);
    }
}

class ContaBancaria {
    private double saldo;

    public ContaBancaria(double saldoInicial) {
        this.saldo = saldoInicial;
    }

    public void saque(double valor) throws SaldoInsuficienteException {
        if (valor > saldo) {
            throw new SaldoInsuficienteException("Saldo insuficiente para realizar o saque.");
        }
        saldo -= valor;
    }

    public double getSaldo() {
        return saldo;
    }
}


public class Excecao {
    public static void main(String[] args) {
        try {
            ContaBancaria conta = new ContaBancaria(100);
            conta.saque(150);
        } catch (SaldoInsuficienteException e) {
            System.out.println(e.getMessage()); 
        }
    }
}

File: listaExercicios/Exceções/excecao.py
class SaldoInsuficienteException(Exception):
    def __init__(self, message="Saldo insuficiente para realizar o saque."):
        self.message = message
        super().__init__(self.message)

class ContaBancaria:
    def __init__(self, saldo_inicial):
        self.saldo = saldo_inicial

    def saque(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteException()
        self.saldo -= valor
        return self.saldo

try:
    conta = ContaBancaria(100)
    conta.saque(150)
except SaldoInsuficienteException as e:
    print(e) 
